### ISSUE TITLE
Nan direction and speed values in velocity feature from example .mat files

### ISSUE THREAD

**Issue by clinzy** [openworm/movement_validation#162]
The example .mat data files I have examined, this includes example_video_feature_file.mat, have Nan for all positions in the speed and direction arrays of the locomotion.velocity features. This is the case for each of the segements (head, midbody, tail, etc). 

>>> test = "/.../example_data/example_video_feature_file.mat"
>>> features = h5py.File(test)
>>> features['worm']['locomotion']['velocity']['head']['speed'][:]
array([[ nan],
       [ nan],
       [ nan],
       ..., 
       [ nan],
       [ nan],
       [ nan]])

**Comment by MichaelCurrie** [comment]
Are you using the latest version of master of movement validation?  Thanks

**Comment by clinzy** [comment]
Yes and the example above is reading in the file directly using h5py.

**Comment by MichaelCurrie** [comment]
Hmm.  That's strange because the worm is definitely moving.  Maybe it's related to the array shape issues I'm getting since some of Jim's code changes went from arrays of shape `(n,)` to `(1,n)`.  (see #163).  In particular, `motion_mode` here is I think being returned with shape `(1,)` incorrectly:

```
Shape mismatch (4642,) vs (1, 4642): locomotion.velocity.head_tip.speed
Values not equal: locomotion.motion_events.forward.num_video_frames
Event mismatch locomotion.motion_events.backward
Shape mismatch (4642,) vs (1,): locomotion.motion_mode
```

Let me try to fix #163 and see if that fixes your issue.

**Comment by JimHokanson** [comment]
@clinzy These values are computed by being smooth over time. Rather than filtering a frames value is calculated based on its neighbors before and after the frame. This means that in the beginning of the video and at the end the values are undefined because no values proceed or follow those frames. It looks like you are only looking at the beginning and end values. If you plot the data you can see that they are not all NaN.
